aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -302,12 +302,33 @@ impl AggregatorData {
     /// delta state.
     /// Note: when we say "aggregator instance" here we refer to Rust struct and
     /// not to the Move aggregator.
-    pub fn get_aggregator(&mut self, id: AggregatorID, limit: u128) -> &mut Aggregator {
-        self.aggregators.entry(id).or_insert_with(|| Aggregator {
-            value: 0,
-            state: AggregatorState::PositiveDelta,
-            limit,
-            history: Some(History::new()),
+    pub fn get_aggregator(
+        &mut self,
+        id: AggregatorID,
+        limit: u128,
+        resolver: &dyn TableResolver,
+        is_aggregator_enabled: bool,
+    ) -> &mut Aggregator {
+        self.aggregators.entry(id).or_insert_with(|| {
+            let (state, value) = if is_aggregator_enabled {
+                (AggregatorState::PositiveDelta, 0)
+            } else {
+                //If aggregator mode is disabled, then read the aggregator value from the table and initialize with it.
+                let key_bytes = id.key.0.to_vec();
+                resolver
+                    .resolve_table_entry(&id.handle, &key_bytes)
+                    .map_err(|_| (AggregatorState::PositiveDelta, 0))
```

Comment:
Would this path be taken if we tried to e.g. add too much to a newly created aggregator in a transaction? (+@georgemitenkov). Maybe we should document it better.

Also I don't follow map_err chained with unwrap, wouldn't unwrap negate any map_err effects? why couldn't we just chain map_or to resolve table_entry?

separately, depending on how the final logic looks like, maybe we can re-use the code w. some other method that needs to read the data.

Reply:
I also don't follow `map_err` and `unwrap`. Let's reuse `read_and_materialize`?

Reply:
Done.